scripts/yocto-build-and-test.sh

```diff
@@ -463,33 +463,44 @@ build_and_test_client() {
 
         cd $BUILDDIR
 
-        # Base image clean
-        clean_build_config
+        # Base image
         bitbake $image_name
-        clean_image=`copy_clean_image "${machine_name}" "${board_name}" "${image_name}" "${device_type}"`
+
+        local images_to_build=$image_name
+
+        if is_building_extra_images_for_board "$board_name"; then
+            if [[ $image_name == core-image-full-cmdline ]]; then
+                images_to_build+=" mender-image-full-cmdline-rofs"
+            fi
+            if has_component monitor-client \
+                    && [[ $image_name == core-image-full-cmdline ]] \
+                    && [[ -f $WORKSPACE/meta-mender/meta-mender-commercial/recipes-extended/images/mender-monitor-image-full-cmdline.bb ]]; then
+                images_to_build+=" mender-monitor-image-full-cmdline"
+            fi
+            if has_component mender-gateway \
+                    && [[ $image_name == core-image-full-cmdline ]] \
+                    && [[ -f $WORKSPACE/meta-mender/meta-mender-commercial/recipes-extended/images/mender-gateway-image-full-cmdline.bb ]]; then
+                images_to_build+=" mender-gateway-image-full-cmdline"
+            fi
+        fi
+
+        bitbake-layers add-layer $WORKSPACE/meta-mender/meta-mender-commercial
+        clean_build_config
+        bitbake $images_to_build
```

Comment:
Is it correct to add `meta-mender-commercial` to build all images? Might be harmless, but this is a change from the current builds, as we will be applying `layer.conf` of this layer to all images (for instance [this line](
https://github.com/mendersoftware/meta-mender/blob/3b876510fbeef9df1b80ee2f9c542b40a5024a81/meta-mender-commercial/conf/layer.conf#L20), but in the future we might add others there and get unexpected results)

Reply:
According to both Drew and Josef, in order for a Yocto layer to be compliant, adding it must have zero effect on the build. So we are actually not compliant, apparently. I'll take this into account and fix this.

Reply:
> According to both Drew and Josef, in order for a Yocto layer to be compliant, adding it must have zero effect on the build. So we are actually not compliant, apparently. I'll take this into account and fix this.

https://github.com/mendersoftware/meta-mender/pull/1763